### PR TITLE
[mailer] - wysiwig editor for templates

### PR DESCRIPTION
**Pull request by nakashu**
- replaced plain codemirror textarea with pell wysiwig editor https://github.com/jaredreich/pell
- codemirror kept for debug/review purpose of the generated html - can be toggled
- ui changes to better use available space on bigger screens
- - combine small fields to lines
- - move html and preview version lower on page
- preview iframe now auto-changes height instead of fixed 1400px
- refactored on-page javascript
- - moved to external file - better utilize vendor resources, can be easily linted and autocorrected now.
- - added some clarifying comments for future generations of developers

notes:
- codemirror is currently unnecessary for this feature to work properly, I suggest to remove it from this page so there is less code to maintain

### CODE REVIEW

File: Mailer/app/presenters/templates/@layout.latte [hunk 1]
```diff
@@ -205,5 +205,6 @@
         {/foreach}
     });
 </script>
+{block javascript}{/block}
```
Comment (rootpd) [inline Mailer/app/presenters/templates/@layout.latte]: Let's use `scripts` as a block label. we already use it in CRM; so we'd stay consistent here.

**Comment by nakashu** [comment]
Will add the requested changes,

What about this one? 
> codemirror is currently unnecessary for this feature to work properly, I suggest to remove it from this page so there is less code to maintain

- should i remove it?
- should it be hidden by default and kept?
- should it be visible by default?

**Comment by rootpd** [comment]
Codemirror should be enabled and visible by default - just please keep some hints in the code how/where to hide and remove it in the future.

The reason is.. the template I attached above. Try to add new paragraph in the visual editor. Whole thing gets screwed up. Unless all is fixed on our side (all the layouts and templates we currently have), we need to keep the editor present.

**Comment by nakashu** [comment]
status:
- codemirror + trumbowyg(wysiwyg) editor available for tamplate editing
- codemirror is default - can be changed in `.env`  - info added to readme
- fields splitted into 2 tabs.
- on 1300x width, preview area increased, to prevent hiding of text
- fixed preview refresh on layout changes

having editor + preview side-by-side on resolutions near 1300 width isn't optimal UX wise, as the editor area is very small. 
Main limitation is the width of preview area and side menu. The preview area is needed at least wide as it is now.

I see three options:
- reduce side menu to hamburger sooner - currently its bellow 1200px (globally or for individual page)
- change layout to have fields below each other - same as when you reduce browser size bellow 1200px width.

**Comment by rootpd** [comment]
Squashed internally, merged via https://github.com/remp2020/remp/commit/8af96a90452d3e43a5dce5683995fbf9a58c66f6.

**Pull request closed without merging**